include/sparrow/utils/repeat_container.hpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <iterator>
+
+namespace sparrow
+{
+    template <class T>
+    class repeat_view
```

Comment:
This should inherit from [view_interface](https://en.cppreference.com/w/cpp/ranges/view_interface) to get all the APIs of an random access range for free.

Reply:
Done